### DISCUSSION TITLE
UI Components

### DISCUSSION THREAD

**Discussion by seanmalbert** [bloom-housing/bloom#1537] [General]
This thread is to discuss updates to UI Components. There is a general need to remove business logic and references to backend types from UI Components. This will make the component library more usable and easier to use across different data types. If, for example, there is a component that accepts a prop of type `Listing`, we should be able to evaluate what interface we actually need for that component and make it generic.

One of the possible end goals of the UI Components is to be able to easily pull it out of the mono repo.

**Comment by seanmalbert** [comment]
@willrlin ,
Aside from this new thread, there are two new issues regarding extending the UI Component tailwind theme and locale strings @ #1538 and #1539.

**Reply by willrlin** [reply]
@elenm @avaleske FYI since you were discussing the styles and string overrides.

**Comment by seanmalbert** [comment]
Updates to UI Components, where business logic is getting stripped out, can be tracked at https://github.com/bloom-housing/bloom/issues?q=label%3A%22UI+Components+-+Strip+Dependencies%22+
